internal/filter/filter.go

```diff
@@ -106,10 +106,27 @@ func (pf *PoolFilter) In(ticket *pb.Ticket) bool {
 		if !ok {
 			return false
 		}
-		// Not simplified so that NaN cases are handled correctly.
-		if !(v >= f.Min && v <= f.Max) {
-			return false
+
+		switch f.Exclude {
```

Comment:
Given this implementation, the testing is not through enough.

There are probably ways to reorder this logic to have fewer statements, but given the complexity of these statements, I think we want full testing on these statements.  That means for each conditional (eg, `v >= f.Min`) testing less than, equal to, and greater than cases.  This would securely lock down the backward compatibility and behavior of these fields.  Please use a helper (like simpleDoubleRange) or other simplification so that we can easily see all of the tests cases are covered and that they are correct.

Reply:
Actually, it might make sense for simpleDoubleRange to take the exclusivity argument.

Reply:
Also, switching to bools would simplify this work (not that we should optimize for writing fewer test cases, but it would show simpler system behavior...) as you could split out the min and max if statements.  However I floated the API choices with a few other people and there's not a clear "this API is better" so I have no preference for which one we put in beta.